### PR TITLE
Student exam

### PR DESCRIPTION
**Pull request by VatsalMewada**
still working...

### CODE REVIEW

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -4,6 +4,11 @@
 class StudentsController < ApplicationController
   before_action :authenticate_user!
   before_action :find_student, only: %i[edit update destroy]
+  def index
+    # byebug
+    @exams = Assign.where(user_id: current_user.id)
```
Comment (swati-bacancy) [inline app/controllers/students_controller.rb]:
```suggestion
    @exams =current_user.assignes
```

File: config/routes.rb [hunk 2]
```diff
@@ -23,6 +31,8 @@
   end
 
   get 'admin/filltable'
+  # get 'student/exam/:id' => 'exams#student_exam', as: 'student_exam'
+  post 'student/result/:id' => 'exams#count_result', as: 'count_result'
```
Comment (swati-bacancy) [inline config/routes.rb]: remove this

File: app/controllers/student_answers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# exam from student side
+class StudentAnswersController < ApplicationController
+  before_action :find_exam, only: %i[new]
```
Comment (swati-bacancy) [inline app/controllers/student_answers_controller.rb]:
```suggestion
  before_action :find_exam, only: :new
```

File: app/helpers/student_answers_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module StudentAnswersHelper
```
Comment (swati-bacancy) [inline app/helpers/student_answers_helper.rb]: remove this file

File: app/views/exams/student_exam.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      <tr>
+        <td><%= count %></td>
+        <td><%= question.info %></td>
+      </tr>
+      <tr>
+        <td></td>
+        <td>
+          <% acount=65 %>
+          <% question.options.each do |option| %>
+            (<%= [acount].pack('U') %>.)
```
Comment (swati-bacancy) [inline app/views/exams/student_exam.html.erb]: what is need of this line?

File: app/views/faculties/_form.html.erb [hunk 6]
```diff
@@ -31,7 +31,7 @@
 
     <div class="form-group row">
       <%= f.label :branch_id, class: "col-md-2" %><br />
-      <%= select_tag(:branch_id,  options_from_collection_for_select(Branch.all, :id, :name), {:class => "col-md-4", :prompt => 'Select Branch', :onchange => "myFunction(this.value);"}) %>
+      <%= f.collection_select(:branch_id, Branch.all, :id, :name, :prompt => 'Select Branch') %>
```
Comment (swati-bacancy) [inline app/views/faculties/_form.html.erb]: use has new syntax

File: app/views/faculties/_form.html.erb [hunk 7]
```diff
@@ -52,7 +52,11 @@
     </div>
     <div class="form-group row">
       <div class="col-md-2"></div>
-      <%= f.submit :class => "col-md-2 text-center" %>
+      <% if @faculty.new_record? %>
+        <%= f.submit formaction: faculties_path %>
```
Comment (swati-bacancy) [inline app/views/faculties/_form.html.erb]:
```suggestion
        <%= f.submit formaction: if @faculty.new_record? ? faculties_path : edit_faculty_path %>
```
user ternary operator

File: config/routes.rb [hunk 8]
```diff
@@ -4,15 +4,23 @@
...
       resources :options
     end
   end
+
+  # resources :questions do
+  #   resources :options
+  # end
+
   resources :students do
     get 'download_excel', on: :collection
```
Comment (swati-bacancy) [inline config/routes.rb]: add this method in resource used in line no 6

File: app/views/exams/student_exam.html.erb (hunk 5 above)
Comment (VatsalMewada) [reply app/views/exams/student_exam.html.erb]: Increment of character

**Pull request merged**